### ISSUE TITLE
TrOCR-based Vision Transformer for Renaissance Spanish OCR (98% accuracy)

### ISSUE THREAD

**Issue by shubhamshukla07** [humanai-foundation/RenAIssance#74]
Hi, I’m Shubham. I’m currently working on the RenAIssance project and exploring an idea around a dynamic, adaptive OCR ensemble for noisy historical documents.

As a first step, I have fine-tuned a TrOCR-based Vision Transformer , the model achieves ~98% character-level accuracy (case-insensitive CER ≈ 1.17%) using beam-search decoding.

<img width="855" height="470" alt="Image" src="https://github.com/user-attachments/assets/fc3c34f6-06c5-4a63-b566-65326d2be62a" />

<img width="842" height="470" alt="Image" src="https://github.com/user-attachments/assets/6714a397-f4c5-4f8b-bab3-c76b5985589e" />

I’m sharing the training curves and qualitative examples here and would appreciate feedback on whether this direction aligns well with the RenAIssance project goals, and whether extending this into an adaptive ensemble framework would be of interest.

**Comment by abhiram123467** [comment]
Hi @shubhamshukla07, great results! 

I've also been working on TrOCR for Renaissance Spanish OCR 
as part of my GSoC 2026 preparation.

My baseline evaluation:
- Tesseract v5 baseline: ~58% CER
- After post-processing: ~44% CER  
- Target with TrOCR fine-tuning: <20% CER

My evaluation repo with full code and analysis:
👉 github.com/abhiram123467/renai-ocr2-test-abhi

A few questions about your approach:
1. Which TrOCR variant did you fine-tune — base or large?
2. What training dataset size did you use?
3. Did you apply any historical Spanish text preprocessing?

Your 98% accuracy is impressive — curious if you tested on 
the BNE dataset or a custom split?

**Comment by shubhamshukla07** [comment]
@abhiram123467 
Hi! Thanks for your interest.
I used TrOCR handwritten-large and fine-tuned it on the Rodrigo dataset. However, the fine-tuning was not done on the full dataset due to time and GPU limitations.

I noticed you're using Tesseract as your baseline, but Tesseract may struggle to achieve high accuracy on complex historical texts, even with post-processing, mainly due to limitations in its architecture.

I’d recommend trying Microsoft TrOCR, which generally performs better on historical or irregular handwriting. However, one important factor is segmentation quality — if the text lines are poorly segmented, or if characters are bleeding/overlapping, it can still significantly hurt the accuracy.
